src/snowflake/snowpark/_internal/telemetry.py

```diff
@@ -341,6 +345,7 @@ def send_upload_file_perf_telemetry(
                 TelemetryField.KEY_CATEGORY.value: TelemetryField.PERF_CAT_UPLOAD_FILE.value,
                 TelemetryField.KEY_FUNC_NAME.value: func_name,
                 TelemetryField.KEY_DURATION.value: duration,
+                TelemetryField.THREAD_IDENTIFIER.value: threading.get_ident(),
```

Comment:
I see we have other telemetry method like `send_function_usage_telemetry`, `send_session_created_telemetry`, do we need to set this field in other methods?

Reply:
actually you bring a great point. I don't think we should update other telemtery but we can add a new telemetery. I can create a new telemetry function `send_new_thread_created` whenever we create a new cursor. This will not be exactly accurate but will give us close to accurate telemetry.